var/spack/repos/builtin/packages/rhash/package.py

```diff
@@ -13,11 +15,23 @@ class Rhash(MakefilePackage):
     homepage = "https://sourceforge.net/projects/rhash/"
     url      = "https://github.com/rhash/RHash/archive/v1.3.5.tar.gz"
 
+    version('1.4.2', sha256='600d00f5f91ef04194d50903d3c79412099328c42f28ff43a0bdb777b00bec62')
     version('1.3.5', sha256='98e0688acae29e68c298ffbcdbb0f838864105f9b2bd8857980664435b1f1f2e')
 
+    # configure: fix clang detection on macOS
+    # Patch accepted and merged upstream, remove on next release
+    patch('https://github.com/rhash/RHash/commit/4dc506066cf1727b021e6352535a8bb315c3f8dc.patch?full_index=1',
+          when='@1.4.2', sha256='3fbfe4603d2ec5228fd198fc87ff3ee281e1f68d252c1afceaa15cba76e9b6b4')
+
     # For macOS build instructions, see:
     # https://github.com/Homebrew/homebrew-core/blob/master/Formula/rhash.rb
 
+    def configure(self, spec, prefix):
+        set_executable('configure')
+        configure_ = Executable('./configure')
```

Comment:
I believe this is already set in `lib/spack/spack/build_environment.py`, you just need to call `configure(...)`